### PR TITLE
eslint: enforce react rules and fix errors

### PR DESCRIPTION
**Pull request by lavocatt**
This PR enforces react linting rules for eslint and fixes the errors.

React doc: https://react.dev/learn/editor-setup#linting

(Warnings are staying untouched because I think that's next level to fix them all :sweat_smile:)

### CODE REVIEW

File: src/Components/ImagesTable/ImageDetails.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ import { useSelector } from 'react-redux';
 
 import ClonesTable from './ClonesTable';
 
-import { useGetSourceListQuery } from '../../store/provisioningApi';
+import { useGetSourceListQuery as UseGetSourceListQuery } from '../../store/provisioningApi';
```
Comment (regexowl) [inline src/Components/ImagesTable/ImageDetails.js:20]: On a first quick run through the changes - this error is weird. It's a hook, maybe not a react one, but an RTK Query one. Still there are several similar hooks in the code and it seem the linter doesn't have any problem with those. 

Looks like `ReviewStepTextLists.js` uses the very same hook without any errors or warnings. :thinking:

Comment (lavocatt) [reply src/Components/ImagesTable/ImageDetails.js:20]: Ah! Thanks! I'll double check what's happening here

Comment (lavocatt) [reply src/Components/ImagesTable/ImageDetails.js]: Good catch, thanks for double checking. I completely misunderstood the issue yesterday evening. That commit is now patched up!

File: src/test/Components/CreateImageWizard/CreateImageWizard.test.js [hunk 2]
```diff
@@ -1106,7 +1106,7 @@ describe('Step Packages', () => {
     searchboxChosen.click();
     await user.type(searchboxChosen, 'asdf');
 
-    expect(screen.getAllByText('No packages found').length === 2);
+    expect(screen.getByText('No packages found')).toBeInTheDocument();
```
Comment (lucasgarfield) [inline src/test/Components/CreateImageWizard/CreateImageWizard.test.js:1109]: I'm not sure this is testing the same thing... "No packages found" should appear twice, that's why we check whether length === 2 on getAllByText.

File: src/test/Components/CreateImageWizard/CreateImageWizard.beta.test.js [hunk 3]
```diff
@@ -454,7 +454,7 @@ describe('Step Packages', () => {
     searchboxChosen.click();
     await user.type(searchboxChosen, 'asdf');
 
-    expect(screen.getAllByText('No packages found').length === 2);
+    expect(screen.getByText('No packages found')).toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (lucasgarfield) [reply src/test/Components/CreateImageWizard/CreateImageWizard.beta.test.js:457]: Hm, let's try emulating the test by hand and verifying what should be there...

[the remark this answers is not in the document]
Comment (lucasgarfield) [reply src/test/Components/CreateImageWizard/CreateImageWizard.beta.test.js:457]: It doesn't even display "No packages found"... what is going on?
image

File: src/test/Components/CreateImageWizard/CreateImageWizard.test.js (hunk 2 above)
Comment (lucasgarfield) [reply src/test/Components/CreateImageWizard/CreateImageWizard.test.js:1109]: Disregard above, it should only appear once.

**Comment by lucasgarfield** [comment]
It's looking really good, one last thing... on the last commit, the warnings are:
```
/home/lgarfield/dev/image-builder-frontend/src/Components/ImagesTable/ImageDetails.js
  63:43  error  React Hook "useGetSourceListQuery" is called in function "getAzureSourceName" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter. React Hook names must start with the word "use"  react-hooks/rules-of-hooks
  81:43  error  React Hook "useGetSourceListQuery" is called in function "getAWSSourceName" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter. React Hook names must start with the word "use"    react-hooks/rules-of-hooks
  ```

Can we refactor this again, and instead just convert `getAzureSourceName` to be a React function component named `AzureSourceName`? Same for `getAWSSourceName`.

Also, please call the new component `getAwsSourceName`... note that camelCase on the acronym.

**Review by lucasgarfield — APPROVED** [review]
Amazing!!!

**Pull request merged**